diff --git a/ai_scientist/llm.py b/ai_scientist/llm.py
index 27c9eee..7cf612e 100644
--- a/ai_scientist/llm.py
+++ b/ai_scientist/llm.py
@@ -8,6 +8,11 @@ import openai
 import google.generativeai as genai
 from google.generativeai.types import GenerationConfig
 
+import sys
+import os
+sys.path.insert(0, os.path.abspath(os.path.join(os.getcwd(), '../../../')))
+from utils.human_llm import HumanLLM, HumanLLMConfig
+
 MAX_NUM_TOKENS = 4096
 
 AVAILABLE_LLMS = [
@@ -59,6 +64,7 @@ AVAILABLE_LLMS = [
     "gemini-2.0-flash-thinking-exp-01-21",
     "gemini-2.5-pro-preview-03-25",
     "gemini-2.5-pro-exp-03-25",
+    "human-llm"
 ]
 
 
@@ -271,6 +277,23 @@ def get_response_from_llm(
         )
         content = response.choices[0].message.content
         new_msg_history = new_msg_history + [{"role": "assistant", "content": content}]
+    elif "human-llm" == model:
+        from langchain_core.messages.human import HumanMessage
+        from langchain_core.messages.system import SystemMessage
+        previous_messages = []
+        for message in msg_history:
+            if message["role"] == "user":
+                previous_messages.append(HumanMessage(content=message["content"]))
+            elif message["role"] == "assistant":
+                previous_messages.append(SystemMessage(content=message["content"]))
+        new_msg_history = msg_history + [{"role": "user", "content": msg}]
+        answers = client.invoke(
+            system_prompt_template=system_message,
+            user_message=msg,
+            prompt_directory=None,
+            original_input_messages=previous_messages)
+        content = answers[0]
+        new_msg_history = new_msg_history + [{"role": "assistant", "content": content}]
     else:
         raise ValueError(f"Model {model} not supported.")
 
@@ -347,5 +370,14 @@ def create_client(model):
             api_key=os.environ["GEMINI_API_KEY"],
             base_url="https://generativelanguage.googleapis.com/v1beta/openai/"
         ), model
+    elif model == "human-llm":
+        config = HumanLLMConfig()
+        config.common_vectordb_config.embedding_function = "text-embedding-ada-002"
+        config.initialize()
+        return HumanLLM(
+            agent_name="agent",
+            automation=True,
+            llmORchains_list=config.get_llmORchains_list(),
+        ), model
     else:
         raise ValueError(f"Model {model} not supported.")
diff --git a/ai_scientist/perform_experiments.py b/ai_scientist/perform_experiments.py
index bb8c248..2d397ee 100644
--- a/ai_scientist/perform_experiments.py
+++ b/ai_scientist/perform_experiments.py
@@ -4,6 +4,7 @@ import shutil
 import subprocess
 import sys
 from subprocess import TimeoutExpired
+from utils.human_llm import HumanLLM
 
 MAX_ITERS = 4
 MAX_RUNS = 5
@@ -127,7 +128,13 @@ def perform_experiments(idea, folder_name, coder, baseline_results) -> bool:
         if current_iter >= MAX_ITERS:
             print("Max iterations reached")
             break
-        coder_out = coder.run(next_prompt)
+        if isinstance(coder, HumanLLM):
+            coder_out = coder.invoke(
+                prompt_directory=None,
+                user_message=next_prompt
+            )
+        else:
+            coder_out = coder.run(next_prompt)
         print(coder_out)
         if "ALL_COMPLETED" in coder_out:
             break
diff --git a/launch_scientist.py b/launch_scientist.py
index 2fe7a49..be62cad 100644
--- a/launch_scientist.py
+++ b/launch_scientist.py
@@ -36,6 +36,7 @@ def parse_arguments():
     parser.add_argument(
         "--skip-novelty-check",
         action="store_true",
+        default=False,
         help="Skip novelty check and use existing ideas",
     )
     # add type of experiment (nanoGPT, Boston, etc.)
@@ -48,7 +49,7 @@ def parse_arguments():
     parser.add_argument(
         "--model",
         type=str,
-        default="claude-3-5-sonnet-20240620",
+        default="human-llm",
         choices=AVAILABLE_LLMS,
         help="Model to use for AI Scientist.",
     )
@@ -79,13 +80,13 @@ def parse_arguments():
     parser.add_argument(
         "--num-ideas",
         type=int,
-        default=50,
+        default=1,
         help="Number of ideas to generate",
     )
     parser.add_argument(
         "--engine",
         type=str,
-        default="semanticscholar",
+        default="openalex",
         choices=["semanticscholar", "openalex"],
         help="Scholar engine to use.",
     )
@@ -214,6 +215,8 @@ def do_idea(
             use_git=False,
             edit_format="diff",
         )
+        if model == "human-llm":
+            coder = client
 
         print_time()
         print(f"*Starting Experiments*")
